### PR TITLE
Update grammar for null aware subscript, plus small fixes.

### PR DESCRIPTION
**Pull request by leafpetersen**

### CODE REVIEW

**Comment by leafpetersen** [comment]
cc @scheglov

File: specification/dartLangSpec.tex [hunk 1]
```diff
@@ -13620,7 +13620,7 @@ \subsection{Assignable Expressions}
 
 <assignableSelector> ::= <unconditionalAssignableSelector>
   \alt `?.' <identifier>
-  \alt `?.[' <expression> `]'
+  \alt `?[' <expression> `]'
```
Comment (lrhn) [inline specification/dartLangSpec.tex]: Grammar should be 
```latex
\alt `?' `[' <expression> `]'
```
with `?` and `[` as separate tokens.

File: accepted/future-releases/nnbd/feature-specification.md [hunk 2]
```diff
@@ -424,7 +437,7 @@ the return type of the getter is not a subtype of the argument type of the
 setter.  Note that this error specifically requires subtyping and not
 assignability and hence makes no exception for `dynamic`.
 
-It is a warning to use a null aware operator (`?.`, `?..`, `??`, `??=`, or
+It is a warning to use a null aware operator (`?.`, `?[`, `?..`, `??`, `??=`, or
```
Comment (lrhn) [inline accepted/future-releases/nnbd/feature-specification.md]: May want to use `?[]` as the name of the operator, to not insinuate that `?[` is a single token.

File: specification/dartLangSpec.tex (hunk 1 above)
Comment (leafpetersen) [reply specification/dartLangSpec.tex]: Done

File: accepted/future-releases/nnbd/feature-specification.md (hunk 2 above)
Comment (leafpetersen) [reply accepted/future-releases/nnbd/feature-specification.md]: Done

**Pull request merged**